File: 594.java
class Solution {
    public int findLHS(int[] nums) {
        // time: O(n), space: O(n)
        Map<Integer, Integer> freqMap = new HashMap<>();
        for (int n : nums) {
            freqMap.put(n, freqMap.getOrDefault(n, 0) + 1);
        }
        int res = 0;
        for (int n : freqMap.keySet()) {
            if (freqMap.containsKey(n + 1)) {
                res = Math.max(res, freqMap.get(n) + freqMap.get(n + 1));
            }
        }
        return res;
    }
}

File: 594.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        # time: O(nlogn), space: O(n)
        cnt = defaultdict(int)
        for n in nums:
            cnt[n] += 1
        res = 0
        sortedCnt = sorted(cnt.items())
        for i in range(len(sortedCnt) - 1):
            if sortedCnt[i][0] == sortedCnt[i + 1][0] - 1:
                res = max(res, sortedCnt[i][1] + sortedCnt[i + 1][1])
        return res